File: 2023/5 If You Give A Seed A Fertilizer/part one.go
package main

func main() {
	content, n, done := getInput()
	if done {
		return
	}

	var lines []string
	var line string

	for i := 0; i < n; i++ {
		if content[i] == '\n' {
			lines = append(lines, line)
			line = ""
		} else {
			line += string(content[i])
		}
	}
}

File: 2021 - Python/2_Dive/part_one.py
def part_one(file_name: str):
    input_file = open(file_name, 'r')
    lines = input_file.readlines()
    depth = 0
    horizontal = 0
    for line in lines:
        words = line.split()
        match words[0]:
            case "forward":
                horizontal += int(words[1])
            case "up":
                depth -= int(words[1])
            case "down":
                depth += int(words[1])
    print("result is " + str(depth * horizontal))
